src/types.ts

```diff
@@ -10,6 +16,31 @@ export type {
   RuntimeSlackFunctionHandler as SlackFunctionHandler,
 } from "./functions/types.ts";
 
+// ----------------------------------------------------------------------------
+// Runtime Types
+// ----------------------------------------------------------------------------
+
+// deno-lint-ignore no-explicit-any
+type AnyCustomType = CustomType<any, any, any>;
+
+type AnyTypedObjectParameterDefinition = TypedObjectParameterDefinition<
+  // deno-lint-ignore no-explicit-any
+  any,
+  // deno-lint-ignore no-explicit-any
+  any
+>;
+
+/**
+ * @description Used to surface runtime Typescript types from Defined Types and Properties
+ */
+export type RuntimeType<
+  T extends
+    | AnyCustomType
+    | AnyTypedObjectParameterDefinition,
+> = T extends AnyCustomType ? FunctionInputRuntimeType<T["definition"]>
+  : T extends AnyTypedObjectParameterDefinition ? FunctionInputRuntimeType<T>
+  : never;
```

Comment:
Do we need the catch-all else case to be never, or could we just have the catch-all directly use the parameter definition? If we do the latter, we don't need to introduce more cases if we decide to support other parameter types.

For instance, I foresee this could eventually become

```ts
export type RuntimeType<
  T extends
    | AnyCustomType
    | AnyTypedObjectParameterDefinition
    | ParameterDefinition
> = T extends AnyCustomType ? FunctionInputRuntimeType<T["definition"]>
  : FunctionInputRuntimeType<T>;
```

Reply:
Tried iterating on this idea but I was not able to make it work, some OAuth types seem to be creating conflicts

Reply:
I've generalized `RuntimeType` to accept any `ParameterDefinition` not only the `TypedObjectParameter` this seems to work as expected

Reply:
Ahhhh I see! Thank you for looking at this. We've found before that the else case of a conditional type does not narrow the available type down based on the check, which means that when we hit `: FunctionInputRuntimeType<T>`, TS still infers T as `T extends ICustomType | ParameterDefinition` rather than just `T extends ParameterDefinition`